tests/func/test_add.py

```diff
@@ -291,6 +300,15 @@ def test(self):
             self.assertEqual(ret, 0)
             self.assertEqual(file_md5_counter.mock.call_count, 3)
 
+            os.rename(self.DATA_DIR, self.DATA_DIR + ".back")
+            ret = main(["checkout"])
+            self.assertEqual(ret, 0)
+            self.assertEqual(file_md5_counter.mock.call_count, 3)
+
+            ret = main(["status"])
+            self.assertEqual(ret, 0)
+            self.assertEqual(file_md5_counter.mock.call_count, 3)
```

Comment:
I wrote this test, so its more of self criticism:
I think we should redesing this test when it comes to file_md5_counter calls:
Insead of asserting count calls, I think we should assert that file_md5 was called for data, data_sub and the temporary file used to calculate dir md5. 
I think that would improve test readibility.
I am not sure we wan't to do it in this task, as its critical, but I think we should make at least task for it.

Reply:
@pared It serves the purpose well enough though. We should probably complement it with some good unit tests for save/checkout methods in Remote at some point, that will make sure that `self.state.save` is called a proper amount of times and with proper arguments. But covering things with proper unit tests is needed for a good chunk of our code base :slightly_frowning_face: